FEXCore/Source/Utils/WritePriorityMutex.h

```diff
@@ -0,0 +1,398 @@
+// SPDX-License-Identifier: MIT
+#pragma once
+#include <atomic>
+#include <cstdint>
+
+#if !defined(_WIN32)
+#include <linux/futex.h> /* Definition of FUTEX_* constants */
+#include <sys/syscall.h> /* Definition of SYS_* constants */
+#include <unistd.h>
+#else
+#include <synchapi.h>
+#endif
+
+#if defined(ASSERTIONS_ENABLED) && ASSERTIONS_ENABLED
+#include <FEXCore/Utils/LogManager.h>
+#endif
+
+#include "Utils/SpinWaitLock.h"
+
+namespace FEXCore::Utils::WritePriorityMutex {
+
+// A mutex that optimizes around the idea that read-locks should be insignificantly low-cost.
+// While a write-lock while rare, should be given priority because its work needs to get in and out to reduce contention on shared locks.
```

Comment:
> // While a write-lock while rare

Looks like a typo. The whole sentence has a bit wonky grammar though, could you double-check this?